.github/workflows/python-package.yml

```diff
@@ -64,3 +66,29 @@ jobs:
           pip install git+https://github.com/${{ github.repository }}.git@${{ github.sha }}
           python examples/testscript.py
           python examples/testscript_multianimal.py
+
+  deploy:
+    # this will run when you have tagged a commit, starting with "v*"
+    # and requires that you have put your twine API key in your
+    # github secrets (see readme for details)
+    needs: [build]
+    runs-on: ubuntu-latest
+    if: contains(github.ref, 'tags')
+    steps:
+      - uses: actions/checkout@v3
+      - name: Set up Python
+        uses: actions/setup-python@v4
+        with:
+          python-version: "3.x"
+      - name: Install dependencies
+        run: |
+          python -m pip install --upgrade pip
+          pip install -U setuptools setuptools_scm wheel twine build
+      - name: Build and publish
+        env:
+          TWINE_USERNAME: __token__
+          TWINE_PASSWORD: ${{ secrets.TWINE_API_KEY }}
+        run: |
+          git tag
+          python -m build .
+          twine upload dist/*
```

Comment:
you could consider pushing to test pypi either on PR merge or when the release PR is tagged --- that way the full workflow ran once before you finalize? but optional ofc, heres an example: https://github.com/AdaptiveMotorControlLab/CEBRA/blob/0378db0b2431d0c50a1e9b80aa1b865869586851/.github/workflows/release-pypi.yml#L34-L43